### PR TITLE
Plebeian Market Nostr identity

### PR DESCRIPTION
**Pull request by ktecho**
Plebeian Market Nostr identity for less tech-savvy users without Nostr extension

### CODE REVIEW

File: api/api.py [hunk 1]
```diff
@@ -215,6 +215,9 @@ def put_me(user):
     if 'stall_description' in request.json:
         user.stall_description = bleach.clean(request.json['stall_description'])
 
+    if 'nostr_private_key' in request.json:
```
Comment (ibz) [inline api/api.py:218]: we should NOT allow users to change the private key!

File: web/src/lib/components/Navbar.svelte [hunk 2]
```diff
@@ -182,6 +182,9 @@
                         </li>
                     {/if}
                     <li class="block md:hidden md:h-0"><a href="/" class="modal-button cursor-pointer">Home</a></li>
+                    {#if ! isProduction() }
+                        <li class="block md:hidden md:h-0"><a href="/townsquare" class="btn btn-ghost normal-case text-xl">Market Square</a></li>
```
Comment (ibz) [inline web/src/lib/components/Navbar.svelte]: /marketsquare?

File: web/src/lib/components/NostrChat.svelte [hunk 3]
```diff
@@ -181,6 +183,11 @@
         }
 
         await updateNostrProfiles();
+
+        while ($user === null) {
```
Comment (ibz) [inline web/src/lib/components/NostrChat.svelte:187]: what are you trying to do here? wait until the user has logged in?

i think this is not the right way to do it.

`user` is a "store" that you can "subscribe" to (don't forget to unsubscribe in onDestroy)! see examples of ".subscribe" in our code.

File: web/src/lib/nostr/pool.ts [hunk 4]
```diff
@@ -72,18 +84,26 @@ export class Pool {
...
-            // ??? event = signEvent(event, nostrPrivateKey)
+            // PM Nostr identity
+            let nostrPrivateKey = user.nostr_private_key;
+
+            if (!nostrPrivateKey) {
+                console.debug('   ** Nostr: Not using extension, but PM identity (private key) not available.')
+                return false;
+            }
+
+            event.sig = signEvent(event, nostrPrivateKey)
```
Comment (ibz) [inline web/src/lib/nostr/pool.ts:95]: ;

File: web/src/lib/nostr/utils.js [hunk 5]
```diff
@@ -59,3 +57,39 @@ export function getChannelIdFromChannelName(channelName) {
...
+            'private': user.nostr_private_key,
+            'public': publicKey,
+            'npub': nip19.npubEncode(publicKey)
+        };
+    }
+
+    let nostrKeys = createNostrKeys();
+    let nostr_private_key = nostrKeys.private;
+
+    putProfile(token, {nostr_private_key},
```
Comment (ibz) [inline web/src/lib/nostr/utils.js:82]: i don't like a few things here:

1) a function called "get..." calls the API with a PUT

and more generally...

2) i don't like the fact that the "nostr utils" package is accessing our API. they should be unrelated!

better to just return something to let the caller (svelte component) know that the profile should be updated.

**Review by ibz — CHANGES_REQUESTED** [review]

File: web/src/lib/components/NostrChat.svelte (hunk 3 above)
Comment (ktecho) [reply web/src/lib/components/NostrChat.svelte:187]: No, I'm waiting until $user is available. It seems it's loaded asynchronously by another component, so sometimes it's `null` and you have to wait a little until it contains the actual user. I did the same in `Login.svelte`.

File: web/src/lib/nostr/utils.js (hunk 5 above)
Comment (ktecho) [reply web/src/lib/nostr/utils.js]: You're right. When I did this a month ago, I was calling this from several places, but anyways it's better in the component.

File: api/api.py (hunk 1 above)
Comment (ktecho) [reply api/api.py:218]: This is needed so the call to `putProfile` works.

File: web/src/lib/components/NostrChat.svelte (hunk 3 above)
Comment (ibz) [reply web/src/lib/components/NostrChat.svelte]: Yes, but I think this is the the best way to do it. `user` is a Svelte store, which has .subscribe, so you can be notified when it changes (and check whether it changed to non-null).

File: api/api.py (hunk 1 above)
Comment (ibz) [reply api/api.py:218]: Ah, right, we are storing the PM-generated key, not the other key. True, sorry...

File: web/src/lib/components/NostrChat.svelte (hunk 3 above)
Comment (ktecho) [reply web/src/lib/components/NostrChat.svelte]: Done. Give it a look.

**Pull request merged**